libkbfs/tlf_edit_history_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+func TestBasicTlfEditHistory(t *testing.T) {
+	var userName1, userName2 libkb.NormalizedUsername = "u1", "u2"
+	config1, _, ctx := kbfsOpsConcurInit(t, userName1, userName2)
+	defer CheckConfigAndShutdown(t, config1)
+
+	clock, now := newTestClockAndTimeNow()
+	config1.SetClock(clock)
+
+	config2 := ConfigAsUser(config1.(*ConfigLocal), userName2)
+	defer CheckConfigAndShutdown(t, config2)
+
+	name := userName1.String() + "," + userName2.String()
+
+	rootNode1 := GetRootNodeOrBust(t, config1, name, false)
+	rootNode2 := GetRootNodeOrBust(t, config2, name, false)
+
+	// user 1 creates a file
+	kbfsOps1 := config1.KBFSOps()
+	_, _, err := kbfsOps1.CreateFile(ctx, rootNode1, "a", false, NoExcl)
+	require.NoError(t, err)
+
+	kbfsOps2 := config2.KBFSOps()
+	err = kbfsOps2.SyncFromServerForTesting(ctx, rootNode2.GetFolderBranch())
+	require.NoError(t, err)
+
+	_, _, err = kbfsOps2.CreateFile(ctx, rootNode2, "b", false, NoExcl)
+	require.NoError(t, err)
+
+	err = kbfsOps1.SyncFromServerForTesting(ctx, rootNode1.GetFolderBranch())
+	require.NoError(t, err)
+
+	_, uid1, err := config1.KBPKI().GetCurrentUserInfo(context.Background())
+	_, uid2, err := config2.KBPKI().GetCurrentUserInfo(context.Background())
+
+	// Each user should see 1 create edit for each user
+	expectedEdits := make(TlfWriterEdits)
+	expectedEdits[uid1] = TlfEditList{{
+		Filepath:  name + "/a",
+		Type:      FileCreated,
+		LocalTime: now,
+	}}
+	expectedEdits[uid2] = TlfEditList{{
+		Filepath:  name + "/b",
+		Type:      FileCreated,
+		LocalTime: now,
+	}}
+
+	edits1, err := kbfsOps1.GetEditHistory(ctx, rootNode1.GetFolderBranch())
+	require.NoError(t, err)
+	edits2, err := kbfsOps2.GetEditHistory(ctx, rootNode2.GetFolderBranch())
+	require.NoError(t, err)
+
+	if !reflect.DeepEqual(expectedEdits, edits1) {
```

Comment:
can do:

`require.Equal(t, expectedEdits, edits1, "User1 has unexpected edit history")`

(it uses DeepEqual and prints out a prettified diff on failure)